core/src/main/java/io/apicurio/hub/api/codegen/jaxrs/OpenApi2CodegenVisitor.java

```diff
@@ -504,34 +503,77 @@ private String methodPath(OpenApiOperation operation) {
         return path;
     }
 
-    private CodegenJavaReturn returnFromSchema(OpenApi31Schema schema) {
+    private void setSchemaProperties(CodegenJavaSchema target, OpenApi31Schema schema) {
         if (schema == null) {
-            return null;
+            return;
         }
-        CodegenJavaReturn cgReturn = new CodegenJavaReturn();
-        cgReturn.setType(null);
-        String $ref = ((Referenceable) schema).get$ref();
+
+        target.setType(null);
+        String $ref = schema.get$ref();
+
         if ($ref != null) {
-            cgReturn.setType(this.typeFromSchemaRef((Document) schema.root(), $ref));
+            target.setType(this.typeFromSchemaRef((Document) schema.root(), $ref));
         } else if ("array".equals(schema.getType())) {
-            cgReturn.setCollection("list");
-            OpenApiSchema items = schema.getItems();
-            CodegenJavaReturn subReturn = this.returnFromSchema((OpenApi31Schema) items);
-            if (subReturn != null && subReturn.getType() != null) {
-                cgReturn.setType(subReturn.getType());
+            if (schema.getItems() != null) {
+                setSchemaProperties(target, schema.getItems());
             }
-            if (subReturn != null && subReturn.getFormat() != null) {
-                cgReturn.setFormat(subReturn.getFormat());
+
+            setIfPresent(schema::isNullable, target::setNullable);
+            setIfPresent(schema::getMaxItems, value -> target.setMaxItems(value.longValue()));
+            setIfPresent(schema::getMinItems, value -> target.setMinItems(value.longValue()));
+
+            if (Boolean.TRUE.equals(schema.isUniqueItems())) {
+                target.setCollection("set");
+            } else {
+                target.setCollection("list");
             }
+        } else if ("object".equals(schema.getType())) {
+            setIfPresent(schema::getType, target::setType);
+            setIfPresent(schema::isNullable, target::setNullable);
+            setIfPresent(schema::getFormat, target::setFormat);
+            // TODO: Consider representing object as map
```

Comment:
would you mind to elaborate?

Reply:
Here I was thinking that if we encounter an `object` type in the parameters, the best way to pass it to the JAX-RS method would be as a Map. The current parameter type would be `java.util.Object` and possibly a `String` at run-time. I left this out of the PR because I thought it may be expanding the scope of the change a bit too much for a single PR.

Reply:
👍